configure.ac

```diff
@@ -1,152 +1,213 @@
+
+# ----------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------
+# Checks for compiler.
+# ----------------------------------------------------------------------
+
+AC_PROG_CC
+AM_PROG_CC_C_O
+AC_PROG_CC_C99
+AS_IF([test "x$ac_cv_prog_cc_c99" = xno], [AC_MSG_ERROR([htop is written in C99. A newer compiler is required])])
+
+# ----------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------
+# Checks for static build.
+# ----------------------------------------------------------------------
+
+AC_ARG_ENABLE([static],
+              [AS_HELP_STRING([--enable-static],
+                              [build a static htop binary @<:@default=no@:>@])],
+              [],
+              [enable_static=no])
+case "$enable_static" in
+   no)
+      ;;
+   yes)
+      AC_DEFINE([BUILD_STATIC], [1], [Define if building static binary.])
+      CFLAGS="$CFLAGS -static"
+      LDFLAGS="$LDFLAGS -static"
+      ;;
+   *)
+      AC_MSG_ERROR([bad value '$enable_static' for --enable-static option])
+      ;;
+esac
+
 # ----------------------------------------------------------------------
-AC_CHECK_LIB([m], [ceil], [], [missing_libraries="$missing_libraries libm"])
 
-# Checks for header files.
+
+# ----------------------------------------------------------------------
+# Checks for generic header files.
 # ----------------------------------------------------------------------
+
 AC_HEADER_DIRENT
 AC_HEADER_STDC
-AC_CHECK_HEADERS([stdlib.h string.h strings.h sys/param.h sys/time.h unistd.h],[:],[
-  missing_headers="$missing_headers $ac_header"
-])
-AC_CHECK_HEADERS([execinfo.h],[:],[:])
+AC_CHECK_HEADERS([ \
+    stdlib.h \
+    string.h \
+    strings.h \
+    sys/param.h \
+    sys/time.h \
+    unistd.h
+   ], [], [AC_MSG_ERROR([can not find required generic header files])])
 
 AC_HEADER_MAJOR
 dnl glibc 2.25 deprecates 'major' and 'minor' in <sys/types.h> and requires to
 dnl include <sys/sysmacros.h>. However the logic in AC_HEADER_MAJOR has not yet
 dnl been updated in Autoconf 2.69, so use a workaround:
 m4_version_prereq([2.70], [],
 [if test "x$ac_cv_header_sys_mkdev_h" != xyes; then
-   AC_CHECK_HEADER(sys/sysmacros.h, [AC_DEFINE(MAJOR_IN_SYSMACROS, 1,
+   AC_CHECK_HEADER([sys/sysmacros.h], [AC_DEFINE([MAJOR_IN_SYSMACROS], [1],
       [Define to 1 if `major', `minor', and `makedev' are declared in <sys/sysmacros.h>.])])
 fi])
 
+# Optional Section
+
+AC_CHECK_HEADERS([execinfo.h])
+
+if test "$my_htop_platform" = darwin; then
+   AC_CHECK_HEADERS([mach/mach_time.h])
+fi
+
+# ----------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------
 # Checks for typedefs, structures, and compiler characteristics.
 # ----------------------------------------------------------------------
-AC_HEADER_STDBOOL
```

Comment:
Why is this check for `stdbool.h` removed, when several of our source files use it (and AFAIK `stdbool.h` is defined in the C99 standard?

Reply:
`AC_HEADER_STDBOOL` only performs checks and defines `HAVE_STDBOOL_H` and `HAVE__BOOL`, but we never actually depend or use these definitions.